Architecture.md

```diff
@@ -65,7 +89,32 @@
 
 #### Client sign-up and login process
 
-***TODO: [CLIENT] mermaid diagram and description of the login process flow, for both the CLI and the web frontend***
+##### Client states
+
+When signing up, the client should provide an email, a username and a password. They can then log into the software as soon as they verify their email. If needed, they can request a password reset using the "forgotten password" process. The account should be either deleted or locked/deactivated after some time of an account being active without a verified email, although this is not implemented for now.
+
+```mermaid
+stateDiagram-v2
+    c : Created
+    v : Verified
+    d : Deleted
+    r : Reset mode
+
+    [*] --> c
+    d --> [*]
+
+    c --> v: email verification
+
+    c --> d: after some time
+
+    v --> r: password reset request
+    r --> v: password change
+
+    v --> d: account deletion request
+
+```
+
```

Comment:
Not sure if we should talk about auth strategies here or in the API section, but I feel that we should say a few words.
```suggestion
##### Authentication strategies

Regarding the authentication strategy, our team first created a `GET /auth/login` endpoint which simply returns a JWT Bearer token to the client. It is then up to the client to store this token and include it in the headers of all requests to protected routes.

Later on, this authentication method was expanded to not only return a token, but also set a JWT token to a client as an HttpOnly cookie in order to keep the user safe from XSS (Cross-Site Scripting) attacks.

Thus, the PaaSTech website no longer has to store a token in its `localStorage`. Nevertheless, we didn't remove the Bearer authentication strategy, so that a user could log in from the CLI as well.

```